pkg/util/kubeconfig.go

```diff
@@ -75,53 +75,50 @@ func getRestConfig(logger logr.Logger, cpName, labelKey, labelValue string) (*re
 	defer cancel()
 
 	kubeClient := *kslclient.GetClient()
-	list := &kfv1aplha1.ControlPlaneList{}
 	labelSelector := labels.SelectorFromSet(labels.Set(map[string]string{
 		ControlPlaneTypeLabel: labelValue,
 	}))
+	var targetCP *kfv1aplha1.ControlPlane
 
 	// wait until there are control planes with supplied labels
-	logger.Info("waiting for cp with label", "key", labelKey, "value", labelValue)
+	logger.Info("Waiting for cp with label", "cpName", cpName, "keyJustKidding", labelKey, "value", labelValue)
 	err := wait.PollUntilContextCancel(ctx, 2*time.Second, true, func(ctx context.Context) (bool, error) {
+		if cpName != "" {
+			cp := &kfv1aplha1.ControlPlane{}
+			err := kubeClient.Get(ctx, client.ObjectKey{Name: cpName}, cp, &client.GetOptions{})
+			if err == nil {
+				targetCP = cp
+				return true, nil
+			}
+			return false, nil
+		}
+		list := &kfv1aplha1.ControlPlaneList{}
```

Comment:
The original intent when this code was initially written was to allow selection of a WDS or ITS either by name or label. If the user does not specify a label, the default label is used. So far, the only usage pattern has been to specify the name for the WDS and just use the default label for the ITS (that is, an ITS name was not required).  Since kubeflex may create also control plane that may not necessarily be used as WDS or ITSs it may still make sense to require to have a label in each to indicate the role. If we relax the restriction on ITSs, we should allow more then one but then we would need to associate WDSs with ITSs.

Reply:
> So far, the only usage pattern has been to specify the name for the WDS and just use the default label for the ITS

We have begun changing this, allowing more than one ITS cp in KubeFlex. @francostellari is working right now on an all-in-one chart that allows a general number of WDSes, a general number of ITSes, and a general selection of ITS for each WDS.

> it may still make sense to require to have a label in each to indicate the role

Indeed it might, perhaps for general orderliness or informing something else. But when the user has indicated the name of the ITS control plane to use, I see no reason to refuse based on labels.